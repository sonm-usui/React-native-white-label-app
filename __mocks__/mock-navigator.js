// @flow
import * as React from 'react'
import { NavigationContainer } from '@react-navigation/native'
import { createStackNavigator } from '@react-navigation/stack'

jest.mock('react-native-screens', () => {
  const RealComponent = jest.requireActual('react-native-screens')
  RealComponent.enableScreens = jest.fn()
  return RealComponent
})

const Stack = createStackNavigator()
export const MockedNavigator = ({
  component,
  params = {},
}: {
  component: React.ElementType,
  params?: Object,
}) => {
  return (
    <NavigationContainer>
      <Stack.Navigator>
        <Stack.Screen name="MockedScreen" initialParams={params}>
          {component}
        </Stack.Screen>
      </Stack.Navigator>
    </NavigationContainer>
  )
}
