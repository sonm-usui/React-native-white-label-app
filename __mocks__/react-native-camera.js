// @flow
import React, { PureComponent } from 'react'

export class RNCamera extends PureComponent<any, any> {
  static Constants = {
    Aspect: {},
    BarCodeType: {},
    Type: { back: 'back', front: 'front' },
    CaptureMode: {},
    CaptureTarget: {},
    CaptureQuality: {},
    Orientation: {},
    FlashMode: {},
    TorchMode: {},
  }

  render() {
    return this.props.children
  }
}

export default RNCamera
