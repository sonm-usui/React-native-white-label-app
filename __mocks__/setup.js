import React from 'react'
import * as ReactNative from 'react-native'
import mockModal from './modal-mock'
import mockView from './view-mock'
import { enableFetchMocks } from 'jest-fetch-mock'

// mock Math.random for <Loader />
const mockMath = Object.create(global.Math)
mockMath.random = () => 0.1
global.Math = mockMath

// window.fetch = fetch
enableFetchMocks()

jest.doMock('react-native', () => {
  // Extend ReactNative
  return Object.setPrototypeOf(
    {
      // ...ReactNative,
      Animated: {
        ...ReactNative.Animated,
      },

      StyleSheet: {
        ...ReactNative.StyleSheet,
      },

      LayoutAnimation: {
        ...ReactNative.LayoutAnimation,
        configureNext: jest.fn(),
      },

      Linking: {
        ...ReactNative.Linking,
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        openURL: jest.fn(() => Promise.resolve()),
        canOpenURL: jest.fn(() => Promise.resolve()),
        getInitialURL: jest.fn(() => Promise.resolve()),
      },

      PixelRatio: {
        get: jest.fn(() => 3), //PixelRatio.get() === 3 then iPhone 6 plus
        roundToNearestPixel: jest.fn((num) => num),
      },

      // Mock a native module
      NativeModules: {
        ...ReactNative.NativeModules,
        BlobModule: {
          ...ReactNative.NativeModules.BlobModule,
          addNetworkingHandler: jest.fn(),
        },
        RNIndy: {
          // Add mock for RNIndy on NativeModules
          // we don't want to mock NativeModules the way we are doing for all above
          // because then we would have to mock everything else as well of NativeModules
          // a lot of other modules depend on it, so we are adding RNIndy
          // which is what we need to mock
          deserializeConnection: jest.fn((_) => Promise.resolve(1)),
          setWalletItem: jest.fn((_) => Promise.resolve(1)),
          deleteWalletItem: jest.fn((_) => Promise.resolve(1)),
          updateWalletItem: jest.fn((_) => Promise.resolve(1)),
          createWalletKey: jest.fn((_) => Promise.resolve('walletKey')),
          setVcxLogger: jest.fn((_) =>
            Promise.resolve('./connectme.rotating.93939939.log')
          ),
          writeToVcxLog: jest.fn(),
          encryptVcxLog: jest.fn(),
          exitAppAndroid: jest.fn(),
          updateMessages: jest.fn(),
        },
        RNCNetInfo: {
          getCurrentState: jest.fn(() => Promise.resolve()),
          addListener: jest.fn(),
          removeListeners: jest.fn(),
        },
        RNPermissions: {},
      },
      Dimensions: {
        ...ReactNative.Dimensions,
        get: jest.fn(() => ({
          width: 320,
          height: 540,
        })),
      },
      UIManager: {
        ...ReactNative.UIManager,
        configureNextLayoutAnimation: jest.fn(),
        getViewManagerConfig: jest.fn(),
      },
      InteractionManager: {
        ...ReactNative.InteractionManager,
        runAfterInteractions(fn) {
          fn()
        },
        createInteractionHandle: jest.fn(),
        clearInteractionHandle: jest.fn(),
      },
      BackHandler: {
        ...ReactNative.BackHandler,
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
      },
      Platform: {
        ...ReactNative.Platform,
      },

      Modal: mockModal,
      View: ReactNative.View,
      Text: ReactNative.Text,
      Image: ReactNative.Image,
      TextInput: ReactNative.TextInput,
    },
    ReactNative
  )
})

jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper')

jest.mock('react-native/Libraries/LogBox/LogBox')

jest.doMock('react-native-vector-icons/lib/icon-button', () => () => {
  const { View, Text } = require('react-native')
  const mockIcon = (props: any) => (
    <View {...props}>
      {props.children !== null ? props.children : <Text>Test children</Text>}
    </View>
  )
})

jest.doMock('react-native-zip-archive', () => ({
  zip: jest.fn(() => Promise.resolve()),
  unzip: jest.fn(() => Promise.resolve()),
}))

jest.doMock('@react-native-firebase/messaging', () => () => ({
  requestPermission: jest.fn(() => Promise.resolve(true)),
  onTokenRefresh: jest.fn(() => Promise.resolve('refreshed-token-value')),
  getToken: jest.fn(() => Promise.resolve('token-value')),
  hasPermission: jest.fn(() => Promise.resolve(true)),
  getInitialNotification: jest.fn(() => Promise.resolve()),
  onNotificationOpenedApp: jest.fn(() => Promise.resolve()),
}))

jest.doMock('@react-native-community/push-notification-ios', () => ({
  removeAllDeliveredNotifications: jest.fn(),
}))

jest.doMock('react-native-sensitive-info', () => {
  // todo: need to handle empty key-chain case
  const secureStorage = {
    ConnectMeKeyChain: {
      identifier: '26u37qFNUk3QwCUQ9ctest',
      phone: '6428900484',
      seed: 'S1StqtsAL4zkFZxmW57sL75zmufmVFQ3',
      pushComMethod: 'dehVox1KRqM:APA91bFWrJea1avml_ELw2MaH60abydtest',
    },
  }
  return {
    setItem: jest.fn((key, value, st) =>
      Promise.resolve((secureStorage['ConnectMeKeyChain'][key] = value))
    ),
    getItem: jest.fn((key) =>
      Promise.resolve(secureStorage['ConnectMeKeyChain'][key])
    ),
    deleteItem: jest.fn((key) =>
      Promise.resolve(delete secureStorage['ConnectMeKeyChain'][key])
    ),
  }
})

jest.mock('@react-native-community/async-storage')

jest.mock('@react-native-community/blur')

jest.mock('@babel/plugin-proposal-decorators')

jest.mock('react-native-modal', () => mockModal)

jest.doMock('rn-fetch-blob', () => ({
  fetch: jest.fn((type, url) => Promise.resolve()),
  fs: {
    dirs: {
      DocumentDir: '/var/application/DocumentDir',
    },
    exists: jest.fn((path) => Promise.resolve(true)),
    unlink: jest.fn((path) => Promise.resolve(true)),
    cp: jest.fn((source, destination) => Promise.resolve(destination)),
  },
}))

jest.mock('react-native-webview')

jest.mock('moment', () =>
  jest.fn((date) => ({
    format(format) {
      if (format === 'MM/DD/YYYY') {
        return '01/10/2018'
      }

      if (format === 'MMMM YYYY') {
        return 'January 2018'
      }

      return '2018-01-10T04:32:43+05:30'
    },
    year(year: number) {
      return this
    },
  }))
)

jest.mock('react-native-share', () => {})

jest.mock('react-native-mail', () => {})

jest.mock('react-native-branch', () => {
  return {
    ADD_TO_CART_EVENT: 'Add to Cart',
    ADD_TO_WISHLIST_EVENT: 'Add to Wishlist',
    PURCHASE_INITIATED_EVENT: 'Purchase Started',
    PURCHASED_EVENT: 'Purchased',
    REGISTER_VIEW_EVENT: 'View',
    SHARE_COMPLETED_EVENT: 'Share Completed',
    SHARE_INITIATED_EVENT: 'Share Started',

    STANDARD_EVENT_ADD_TO_CART: 'ADD_TO_CART',
    STANDARD_EVENT_ADD_TO_WISHLIST: 'ADD_TO_WISHLIST',
    STANDARD_EVENT_VIEW_CART: 'VIEW_CART',
    STANDARD_EVENT_INITIATE_PURCHASE: 'INITIATE_PURCHASE',
    STANDARD_EVENT_ADD_PAYMENT_INFO: 'ADD_PAYMENT_INFO',
    STANDARD_EVENT_PURCHASE: 'PURCHASE',
    STANDARD_EVENT_SPEND_CREDITS: 'SPEND_CREDITS',

    STANDARD_EVENT_SEARCH: 'SEARCH',
    STANDARD_EVENT_VIEW_ITEM: 'VIEW_ITEM',
    STANDARD_EVENT_VIEW_ITEMS: 'VIEW_ITEMS',
    STANDARD_EVENT_RATE: 'RATE',
    STANDARD_EVENT_SHARE: 'SHARE',

    STANDARD_EVENT_COMPLETE_REGISTRATION: 'COMPLETE_REGISTRATION',
    STANDARD_EVENT_COMPLETE_TUTORIAL: 'COMPLETE_TUTORIAL',
    STANDARD_EVENT_ACHIEVE_LEVEL: 'ACHIEVE_LEVEL',
    STANDARD_EVENT_UNLOCK_ACHIEVEMENT: 'UNLOCK_ACHIEVEMENT',

    // don't need any type checks for mocks, because they are not used
    // for type checking in code
    subscribe(cb) {},
  }
})

jest.mock('react-native-splash-screen', () => ({
  hide: jest.fn(),
}))

jest.mock('react-native-unique-id', () =>
  jest.fn(() => Promise.resolve('uniqueDeviceId'))
)

jest.mock('react-native-image-crop-picker', () => ({
  openPicker: jest.fn(() =>
    Promise.resolve({ path: '/var/application/DocumentDir/user-avatar.jpeg' })
  ),
}))

jest.mock('react-native-device-info', () => ({
  getModel: jest.fn(),
}))

jest.mock('react-native-version-number', () => ({
  appVersion: '1.0',
  buildVersion: '500',
}))

jest.mock('apptentive-react-native', () => ({
  Apptentive: {
    register: () => Promise.resolve(null),
    presentMessageCenter: jest.fn(),
    onAuthenticationFailed: (reason) => null,
  },
  ApptentiveConfiguration: () => null,
}))

jest.mock('react-native-gesture-handler', () => {
  const View = require('react-native/Libraries/Components/View/View')

  return {
    Swipeable: View,
    DrawerLayout: View,
    State: {
      END: 'END',
    },
    ScrollView: View,
    Slider: View,
    Switch: View,
    TextInput: View,
    ToolbarAndroid: View,
    ViewPagerAndroid: View,
    DrawerLayoutAndroid: View,
    WebView: View,
    NativeViewGestureHandler: View,
    TapGestureHandler: View,
    FlingGestureHandler: View,
    ForceTouchGestureHandler: View,
    LongPressGestureHandler: View,
    PanGestureHandler: View,
    PinchGestureHandler: View,
    RotationGestureHandler: View,
    /* Buttons */
    RawButton: View,
    BaseButton: View,
    RectButton: View,
    BorderlessButton: View,
    /* Other */
    FlatList: View,
    gestureHandlerRootHOC: jest.fn(),
    Directions: {},
  }
})

jest.mock('react-native-snackbar', () => ({
  show: jest.fn(),
  LENGTH_LONG: 3,
}))

jest.mock('reactotron-react-native', () => ({
  configure: () => ({ useReactNative: () => ({ connect() {} }) }),
}))

jest.mock('react-native-document-picker', () => ({
  show: (config: Object, callback: (error: any, result: any) => void) => {
    callback(null, {
      fileName: 'test-file-name',
      fileSize: 1200,
      type: 'zip',
      uri: 'file:///document-directory',
    })
  },
}))

jest.mock('react-native-shake')

jest.mock('react-native-screens', () => ({
  enableScreens: jest.fn(),
}))

jest.mock('react-native-file-viewer', () => ({
  open: jest.fn((path, options) => Promise.resolve()),
}))

jest.mock('react-native-reanimated', () =>
  jest.requireActual('../node_modules/react-native-reanimated/mock')
)

jest.mock('react-native-redash/lib/module/v1', () => ({
  min: jest.fn(),
  mix: jest.fn(),
  snapPoint: jest.fn(),
  timing: jest.fn(),
  usePanGestureHandler: jest.fn(),
  useTransition: jest.fn(),
  useValue: jest.fn(),
}))
