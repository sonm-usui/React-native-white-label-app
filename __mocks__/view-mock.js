// @flow
import React, { PureComponent } from 'react'
import { View } from 'react-native'

// We don't need to type check external library
// and we don't need types for mock test stub
class MockView extends PureComponent<any, void> {
  render() {
    return <View {...this.props}>{this.props.children}</View>
  }
}

export default MockView
