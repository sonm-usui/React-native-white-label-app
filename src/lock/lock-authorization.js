// @flow
/**
 * Intend to verify user actions
 * we will ask authorize user for some actions in app
 * either by asking for TouchId or asking user to enter pin code
 */
import React, { useCallback } from 'react'
import { View, StyleSheet } from 'react-native'

import type { LockAuthorizationProps } from './type-lock'

import LockEnter from './lock-enter'
import { lockAuthorizationHomeRoute } from '../common'
import { useInteractionDone } from '../hooks/use-interactions-done'
import { LoaderGif } from '../components/loader-gif/loader-gif'
import { colors } from '../common/styles/constant'
import { headerDefaultOptions } from '../navigation/navigation-header-config'

export const LockAuthorization = ({ route }: LockAuthorizationProps) => {
  const [interactionDone] = useInteractionDone()
  const onSuccess = useCallback(() => {
    const { params } = route
    params && params.onSuccess && params.onSuccess()
  }, [])

  return !interactionDone ? (
    LoaderGif
  ) : (
    <View style={styles.container}>
      <LockEnter onSuccess={onSuccess} />
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    width: '100%',
    height: '100%',
    backgroundColor: colors.white,
  },
})

export const lockAuthorizationScreen = {
  routeName: lockAuthorizationHomeRoute,
  screen: LockAuthorization,
  options: headerDefaultOptions({
    headline: undefined,
    headerHideShadow: true,
    transparent: false,
  }),
}
