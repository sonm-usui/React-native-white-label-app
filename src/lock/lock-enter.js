// @flow
import React, { Component } from 'react'
import { StyleSheet, Platform, Image, Text, View } from 'react-native'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'

import {
  Container,
  CustomText,
  PinCodeBox,
  CustomView,
  Icon,
} from '../components'
import {
  color,
  colors,
  fontFamily,
  OFFSET_3X,
  OFFSET_4X,
} from '../common/styles'
import { CHECK_PIN_IDLE, CHECK_PIN_SUCCESS, CHECK_PIN_FAIL } from './type-lock'
import type { LockEnterState, LockEnterProps } from './type-lock'
import {
  checkPinAction,
  checkPinStatusIdle,
  putPinFailData,
} from './lock-store'
import { switchErrorAlerts } from '../store/config-store'
import type { Store } from '../store/type-store'
import { ENTER_YOUR_PASS_CODE_MESSAGE } from '../common/message-constants'
import { isBiggerThanShortDevice } from '../common/styles/constant'
import { verticalScale } from 'react-native-size-matters'
import { appName, LockHeader } from '../external-imports'

const lockImage = require('../images/lockCombo.png')
const backgroundImg = require('../images/wave1.png')

const styles = StyleSheet.create({
  //TODO : add animations when keyboard popups.
  text: {
    marginTop: OFFSET_4X,
    minHeight: OFFSET_3X,
    marginBottom: OFFSET_4X,
  },
  titleText: {
    fontFamily,
    fontSize: 26,
    fontStyle: 'normal',
    letterSpacing: 0.5,
    lineHeight: 31,
    minHeight: verticalScale(62),
    fontWeight: '700',
  },
})

export const WrongPinText = (
  <CustomText style={[styles.titleText]} center h4 bg="tertiary" tertiary thick>
    Wrong passcode! Please try again
  </CustomText>
)

export class LockEnter extends Component<LockEnterProps, LockEnterState> {
  pinCodeBox = null
  clearFailStatus = () => {
    this.props.checkPinStatusIdle()
  }

  clearFailStatusDelayed = () => {
    setTimeout(this.clearFailStatus, 1000)
  }

  onPinComplete = (pin: string) => {
    this.props.checkPinAction(pin, this.props.isAppLocked)
  }

  componentDidUpdate(prevProps: LockEnterProps) {
    if (prevProps.shouldLockApp !== this.props.shouldLockApp) {
      this.props.putPinFailData()

      if (!this.props.shouldLockApp) {
        this.pinCodeBox && this.pinCodeBox.showKeyboard()
      }
    }

    if (this.props.checkPinStatus !== prevProps.checkPinStatus) {
      if (this.props.checkPinStatus === CHECK_PIN_SUCCESS) {
        this.pinCodeBox && this.pinCodeBox.hideKeyboard()
        this.props.onSuccess()

        this.pinCodeBox && this.pinCodeBox.clear && this.pinCodeBox.clear()
      } else if (this.props.checkPinStatus === CHECK_PIN_FAIL) {
        this.pinCodeBox && this.pinCodeBox.clear && this.pinCodeBox.clear()
        // set status back to idle so we can come to this else again
        this.clearFailStatusDelayed()
      }
    }
  }

  componentDidMount() {
    this.props.putPinFailData()
    if (this.props.checkPinStatus === CHECK_PIN_SUCCESS) {
      this.clearFailStatus()
    }
  }

  renderFailMessages = () => {
    if (this.props.shouldLockApp && this.props.lockdownTimeMessage) {
      return (
        <CustomView center>
          <Text style={stylesRecovery.failedAttemptsText}>
            {this.props.numberOfAttemptsMessage}
          </Text>
          <Text style={stylesRecovery.failedAttemptsText}>
            {this.props.lockdownTimeMessage}
          </Text>
        </CustomView>
      )
    } else if (
      !this.props.shouldLockApp &&
      this.props.numberOfAttemptsMessage
    ) {
      return (
        <CustomView center>
          <Text style={stylesRecovery.failedAttemptsText}>
            {this.props.numberOfAttemptsMessage}
          </Text>
        </CustomView>
      )
    } else return null
  }

  render() {
    const {
      checkPinStatus,
      message = ENTER_YOUR_PASS_CODE_MESSAGE,
    } = this.props

    const EnterPinText = (
      <CustomText
        style={[styles.titleText]}
        center
        h4
        bg="tertiary"
        tertiary
        thick
        testID="pass-code-input-text"
      >
        {message}
      </CustomText>
    )

    return (
      <Container>
        {this.props.fromRecovery ? (
          <Container safeArea fifth>
            <CustomView center>
              {LockHeader ? <LockHeader /> : <View />}
            </CustomView>
            <Image
              source={backgroundImg}
              style={[stylesRecovery.backgroundImg]}
            />
            <CustomView
              center
              transparentBg
              style={[stylesRecovery.topLockIcon]}
            >
              <CustomView style={[stylesRecovery.blackStrip]} />
              <CustomView style={[stylesRecovery.lockIconWrapper]}>
                <Icon extraLarge halo src={lockImage} />
              </CustomView>
            </CustomView>
            <CustomView center verticalSpace>
              <CustomText
                center
                transparentBg
                style={[stylesRecovery.lockHeading]}
                heavy
                charcoal
              >
                Please enter your current {appName} passcode!
              </CustomText>
            </CustomView>
            <CustomView center>
              <PinCodeBox
                ref={(pinCodeBox) => {
                  this.pinCodeBox = pinCodeBox
                }}
                onPinComplete={this.onPinComplete}
              />
            </CustomView>
            <CustomView center doubleVerticalSpace>
              <CustomText
                transparentBg
                center
                h6
                bold
                style={[stylesRecovery.newPasscodeText]}
                onPress={this.props.setupNewPassCode}
                testID={'set-up-new-passcode-recovery'}
              >
                Or Set Up New Passcode
              </CustomText>
            </CustomView>
          </Container>
        ) : (
          <Container tertiary>
            <CustomView center>
              {LockHeader ? <LockHeader /> : <View />}
            </CustomView>
            <CustomView
              style={[styles.text]}
              center
              onPress={this.props.switchErrorAlerts}
              testID="pin-code-input-boxes"
            >
              {checkPinStatus === CHECK_PIN_IDLE && EnterPinText}
              {checkPinStatus === CHECK_PIN_SUCCESS && EnterPinText}
              {checkPinStatus === CHECK_PIN_FAIL && WrongPinText}
            </CustomView>
            <CustomView center>
              {this.props.isAppLocked && this.renderFailMessages()}
              <PinCodeBox
                ref={(pinCodeBox) => {
                  this.pinCodeBox = pinCodeBox
                }}
                onPinComplete={this.onPinComplete}
                enableCustomKeyboard={this.props.enableCustomKeyboard}
                disableKeyboard={this.props.shouldLockApp}
              />
            </CustomView>
          </Container>
        )}
      </Container>
    )
  }
}

const stylesRecovery = StyleSheet.create({
  failedAttemptsText: {
    color: colors.red,
    fontSize: 17,
    fontWeight: '500',
    fontFamily: fontFamily,
  },
  backgroundImg: {
    position: 'absolute',
    transform: [{ rotate: '-180deg' }],
    marginTop: '-92%',
    width: '100%',
  },
  topLockIcon: {
    ...Platform.select({
      ios: {
        marginTop: isBiggerThanShortDevice ? '14%' : '3%',
      },
      android: {
        marginTop: '8%',
      },
    }),
  },
  blackStrip: {
    position: 'absolute',
    height: 8,
    backgroundColor: colors.gray1,
    width: '100%',
  },
  lockIconWrapper: {
    backgroundColor: colors.gray1,
    borderRadius: 50,
  },
  lockHeading: {
    fontSize: isBiggerThanShortDevice ? 23 : 16,
    width: '80%',
    ...Platform.select({
      ios: {
        marginTop: isBiggerThanShortDevice ? '7%' : '2%',
        marginBottom: '-1%',
      },
      android: {
        marginTop: isBiggerThanShortDevice ? '2%' : 0,
      },
    }),
  },
  newPasscodeText: {
    ...Platform.select({
      ios: {
        marginTop: isBiggerThanShortDevice ? '6.1%' : '1%',
      },
      android: {
        marginTop: isBiggerThanShortDevice ? '2%' : 0,
      },
    }),
    color: color.bg.fifteenth.color,
  },
})

const mapStateToProps = (state: Store) => ({
  isAppLocked: state.lock.isAppLocked,
  checkPinStatus: state.lock.checkPinStatus,
  shouldLockApp: state.lock.shouldLockApp,
  numberOfAttemptsMessage: state.lock.numberOfAttemptsMessage,
  lockdownTimeMessage: state.lock.lockdownTimeMessage,
})

const mapDispatchToProps = (dispatch) =>
  bindActionCreators(
    {
      checkPinAction,
      checkPinStatusIdle,
      switchErrorAlerts,
      putPinFailData,
    },
    dispatch
  )

export default connect(mapStateToProps, mapDispatchToProps)(LockEnter)
