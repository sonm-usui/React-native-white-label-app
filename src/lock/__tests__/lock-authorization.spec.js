// @flow
import React from 'react'
import { create, act } from 'react-test-renderer'
import { Provider } from 'react-redux'
import { LockAuthorization } from '../lock-authorization'
import { getNavigation, getStore } from '../../../__mocks__/static-data'
import { LockEnter } from '../lock-enter'

jest.useFakeTimers()

describe('<LockAuthorization />', () => {
  it('should match snapshot', () => {
    const { component } = setup()
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('go back and call success onSuccess', () => {
    const { props, lockEnterInstance } = setup()

    lockEnterInstance && lockEnterInstance.props.onSuccess()
    expect(props.route.params.onSuccess).toHaveBeenCalled()
  })

  function setup() {
    const props = getProps()
    const store = getStore()
    let component
    act(() => {
      component = create(
        <Provider store={store}>
          <LockAuthorization {...props} />
        </Provider>
      )
    })

    act(() => {
      jest.advanceTimersByTime(1000)
    })
    if (!component) {
      throw new Error('component not found')
    }

    const lockEnterInstance = component.root.findByType(LockEnter).instance

    return { props, component, lockEnterInstance }
  }

  function getProps() {
    return {
      navigation: getNavigation(),
      route: {
        params: { onSuccess: jest.fn(), onAvoid: jest.fn() },
      },
    }
  }
})
