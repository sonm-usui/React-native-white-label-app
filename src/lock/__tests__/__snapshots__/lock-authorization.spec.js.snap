// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LockAuthorization /> should match snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "height": "100%",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "flex": 1,
        }
      }
      tertiary={true}
    >
      <View
        center={true}
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      />
      <View
        accessibilityLabel="pin-code-input-boxes-touchable"
        accessible={false}
        center={true}
        focusable={true}
        hitSlop={
          Object {
            "bottom": 20,
            "left": 20,
            "right": 20,
            "top": 20,
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginBottom": 40,
            "marginTop": 40,
            "minHeight": 30,
          }
        }
        testID="pin-code-input-boxes-touchable"
      >
        <Text
          accessibilityLabel="Enter your passcode"
          accessible={true}
          allowFontScaling={false}
          style={
            Array [
              Object {
                "fontSize": 19,
              },
              Object {
                "fontFamily": "System",
              },
              Object {
                "backgroundColor": "#ffffff",
                "color": "#777777",
              },
              Object {
                "fontWeight": "800",
              },
              Object {
                "textAlign": "center",
              },
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              Object {
                "fontFamily": "System",
                "fontSize": 26,
                "fontStyle": "normal",
                "fontWeight": "700",
                "letterSpacing": 0.5,
                "lineHeight": 31,
                "minHeight": 49.23529411764706,
              },
            ]
          }
          suppressHighlighting={true}
          testID="pass-code-input-text"
        >
          Enter your passcode
        </Text>
      </View>
      <View
        center={true}
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <View
          style={Object {}}
        >
          <View
            accessible={true}
            focusable={true}
            hitSlop={
              Object {
                "bottom": 20,
                "left": 20,
                "right": 20,
                "top": 20,
              }
            }
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            row={true}
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              accessibilityLabel="pin-code-digit-0-touchable"
              accessible={false}
              center={true}
              focusable={true}
              hitSlop={
                Object {
                  "bottom": 20,
                  "left": 20,
                  "right": 20,
                  "top": 20,
                }
              }
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "height": 66,
                  "justifyContent": "center",
                  "marginHorizontal": 5,
                  "width": 38,
                }
              }
              testID="pin-code-digit-0-touchable"
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#EAEAEA",
                    "borderRadius": 20,
                    "height": 26,
                    "width": 26,
                  }
                }
              />
            </View>
            <View
              accessibilityLabel="pin-code-digit-1-touchable"
              accessible={false}
              center={true}
              focusable={true}
              hitSlop={
                Object {
                  "bottom": 20,
                  "left": 20,
                  "right": 20,
                  "top": 20,
                }
              }
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "height": 66,
                  "justifyContent": "center",
                  "marginHorizontal": 5,
                  "width": 38,
                }
              }
              testID="pin-code-digit-1-touchable"
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#EAEAEA",
                    "borderRadius": 20,
                    "height": 26,
                    "width": 26,
                  }
                }
              />
            </View>
            <View
              accessibilityLabel="pin-code-digit-2-touchable"
              accessible={false}
              center={true}
              focusable={true}
              hitSlop={
                Object {
                  "bottom": 20,
                  "left": 20,
                  "right": 20,
                  "top": 20,
                }
              }
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "height": 66,
                  "justifyContent": "center",
                  "marginHorizontal": 5,
                  "width": 38,
                }
              }
              testID="pin-code-digit-2-touchable"
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#EAEAEA",
                    "borderRadius": 20,
                    "height": 26,
                    "width": 26,
                  }
                }
              />
            </View>
            <View
              accessibilityLabel="pin-code-digit-3-touchable"
              accessible={false}
              center={true}
              focusable={true}
              hitSlop={
                Object {
                  "bottom": 20,
                  "left": 20,
                  "right": 20,
                  "top": 20,
                }
              }
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "height": 66,
                  "justifyContent": "center",
                  "marginHorizontal": 5,
                  "width": 38,
                }
              }
              testID="pin-code-digit-3-touchable"
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#EAEAEA",
                    "borderRadius": 20,
                    "height": 26,
                    "width": 26,
                  }
                }
              />
            </View>
            <View
              accessibilityLabel="pin-code-digit-4-touchable"
              accessible={false}
              center={true}
              focusable={true}
              hitSlop={
                Object {
                  "bottom": 20,
                  "left": 20,
                  "right": 20,
                  "top": 20,
                }
              }
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "height": 66,
                  "justifyContent": "center",
                  "marginHorizontal": 5,
                  "width": 38,
                }
              }
              testID="pin-code-digit-4-touchable"
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#EAEAEA",
                    "borderRadius": 20,
                    "height": 26,
                    "width": 26,
                  }
                }
              />
            </View>
            <View
              accessibilityLabel="pin-code-digit-5-touchable"
              accessible={false}
              center={true}
              focusable={true}
              hitSlop={
                Object {
                  "bottom": 20,
                  "left": 20,
                  "right": 20,
                  "top": 20,
                }
              }
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "height": 66,
                  "justifyContent": "center",
                  "marginHorizontal": 5,
                  "width": 38,
                }
              }
              testID="pin-code-digit-5-touchable"
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#EAEAEA",
                    "borderRadius": 20,
                    "height": 26,
                    "width": 26,
                  }
                }
              />
            </View>
          </View>
          <TextInput
            accessibilityLabel="pin-code-input-box"
            accessible={true}
            autoCorrect={false}
            blurOnSubmit={false}
            editable={true}
            enablesReturnKeyAutomatically={false}
            keyboardAppearance="dark"
            keyboardType="number-pad"
            maxLength={6}
            onChangeText={[Function]}
            style={
              Object {
                "height": 0,
                "position": "absolute",
                "right": -999,
                "width": 0,
              }
            }
            testID="pin-code-input-box"
            value=""
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;
